src/renderer/component/selectChannel/view.jsx

```diff
@@ -0,0 +1,221 @@
+// @flow
+import React from 'react';
+import { isNameValid } from 'lbryURI';
+import { FormRow, FormField } from 'component/common/form';
+import { BusyMessage } from 'component/common';
+import Link from 'component/link';
+
+type Props = {
+  channel: string, //currently selected channel
+  channels: Array<{ name: string }>,
+  balance: number,
+  onChannelChange: (string) => void,
+  createChannel: (string, number) => Promise<any>,
+  fetchChannelListMine: () => void,
+  fetchingChannels: boolean,
+}
+
+type State = {
+  newChannelName: string,
+  newChannelBid: number,
+  addingChannel: boolean,
+  creatingChannel: boolean,
+  newChannelNameError: string,
+  newChannelBidError: string,
+  createChannelError: ?string
+}
+
+class ChannelSection extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      newChannelName: '',
+      newChannelBid: 10,
+      addingChannel: false,
+      creatingChannel: false,
+      newChannelNameError: '',
+      newChannelBidError: '',
+      createChannelError: undefined,
+    };
+
+    (this: any).handleChannelChange = this.handleChannelChange.bind(this);
+    (this: any).handleNewChannelNameChange = this.handleNewChannelNameChange.bind(this);
+    (this: any).handleNewChannelBidChange = this.handleNewChannelBidChange.bind(this);
+    (this: any).handleCreateChannelClick = this.handleCreateChannelClick.bind(this);
+  }
+
+  componentDidMount() {
+    const { channels, fetchChannelListMine, fetchingChannels } = this.props;
+    if (!channels.length && !fetchingChannels) {
+      fetchChannelListMine();
+    }
+  }
+
+  handleChannelChange(event: SyntheticInputEvent<*>) {
+    const { onChannelChange } = this.props;
+    const channel = event.target.value;
+
+    if (channel === 'new') {
+      this.setState({ addingChannel: true });
```

Comment:
Now that the publish page generally retains state via Redux, will this internal component state be a potential source of error or confusion?

Reply:
This might be a good component to test out your idea, just to see how it works in practice. I'll keep it as is for now, but look into refactoring when we add it to the "Your publishes" page, or somewhere else.

Reply:
:+1: